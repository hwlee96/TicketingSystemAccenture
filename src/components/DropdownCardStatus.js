import React, { Component } from "react";
import "../styles/App.css";
import UpdateButton from "./UpdateButton";

import arrow from "../Resources/Icons/iconfinder_icon-ios7-arrow-down_211687.svg";

class DropdownCardStatus extends Component {
  constructor() {
    super();

    this.state = {
      showMenu: false
    };

    this.showMenu = this.showMenu.bind(this);
    this.closeMenu = this.closeMenu.bind(this);
  }

  showMenu(event) {
    event.preventDefault();

    this.setState({ showMenu: true }, () => {
      document.addEventListener("click", this.closeMenu);
    });
  }

  closeMenu(event) {
    if (!this.dropdownMenu.contains(event.target)) {
      this.setState({ showMenu: false }, () => {
        document.removeEventListener("click", this.closeMenu);
      });
    }
  }

  render() {
    let pageId = this.props.idd;
    // console.log("pageID is: " + pageId);

    return (
      <div>
        <button
          onClick={this.showMenu}
          id="transparentButton"
          className="transparentButton-status"
        >
          <img src={arrow} className="arrow-down-2" alt="arrow" />
        </button>

        {this.state.showMenu ? (
          <div
            className="menu"
            ref={element => {
              this.dropdownMenu = element;
            }}
          >
            <ul className="dropdownList">
              <li className="buttonList">
                <UpdateButton
                  idd={pageId}
                  text="Open"
                  logic="status"
                  isClient={this.props.isClient}
                />
              </li>
              <li className="buttonList">
                <UpdateButton
                  idd={pageId}
                  text="Resolved"
                  logic="status"
                  isClient={this.props.isClient}
                />
              </li>
            </ul>
          </div>
        ) : null}
      </div>
    );
  }
}

export default DropdownCardStatus;
