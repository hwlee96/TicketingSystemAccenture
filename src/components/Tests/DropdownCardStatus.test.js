import React from "react";
import renderer from "react-test-renderer";
import { shallow, mount, render, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import DropdownCardStatus from "../DropdownCardStatus";

configure({ adapter: new Adapter() });

it("renders correctly", () => {
  const tree = renderer.create(<DropdownCardStatus />).toJSON();
  expect(tree).toMatchSnapshot();
});
