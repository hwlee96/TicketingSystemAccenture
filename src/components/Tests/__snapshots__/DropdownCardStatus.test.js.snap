// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <button
    className="transparentButton-status"
    id="transparentButton"
    onClick={[Function]}
  >
    <img
      alt="arrow"
      className="arrow-down-2"
      src="iconfinder_icon-ios7-arrow-down_211687.svg"
    />
  </button>
</div>
`;
